modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py

```diff
@@ -375,22 +361,28 @@ def _apply_func(partition, func, *args, **kwargs):  # pragma: no cover
     str
         The node IP address of the worker process.
     """
+    logger = get_worker_logger(JOB_ID)
+    logger.info(
+        f"START::PANDAS-API::PandasOnRayDataframePartition._apply_func {func=} {args=} {kwargs=}"
+    )
     try:
         result = func(partition, *args, **kwargs)
     # Sometimes Arrow forces us to make a copy of an object before we operate on it. We
     # don't want the error to propagate to the user, and we want to avoid copying unless
     # we absolutely have to.
     except ValueError:
         result = func(partition.copy(), *args, **kwargs)
-    return (
+    res = (
         result,
         len(result) if hasattr(result, "__len__") else 0,
         len(result.columns) if hasattr(result, "columns") else 0,
         get_node_ip_address(),
     )
+    logger.info(f"STOP::PANDAS-API::PandasOnRayDataframePartition._apply_func")
+    return res
```

Comment:
you can keep `return` statement and add logging by using pattern like
```python
try:
    return foo
finally:
    logger.info("bar")
```
it's also better in terms of any exception raised during `return foo` line (in our case, for example, an exception in `get_node_ip_address()`) would _not_ prevent logging `STOP` message